test/16_recode.jl

```diff
@@ -534,4 +538,70 @@ end
     @test !isordered(y)
 end
 
+@testset "replace with CategoricalArray" begin
+    function testf(f, T, x::CategoricalArray, y, pairs::Pair...)
+        ca = f(x, pairs...)
+        @test ca ≅ y
+        @test ca isa CategoricalArray{T}
+
+        if VERSION >= v"0.7.0-"
+            a = f(Array(x), pairs...)
+            @test a ≅ y
+            @test a isa Array{T}
+        end
+
+        ca
+    end
+
+    @testset "string with missings" begin
+        x = categorical(["a", "b", missing, "a"])
+
+        testf(replace, String, x, ["a", "b", "", "a"], missing => "")
+        testf(replace, Union{String, Missing}, x, ["a", "c", missing, "a"], "b" => "c")
+        testf(replace, Any, x, [1, 2, missing, 1], "a" => 1, "b" => 2)
+        testf(replace, Any, x, [1, 2, 3, 1], "a" => 1, "b" => 2, missing => 3)
+        y = testf(replace!, Union{String, Missing}, x, ["a", "c", missing, "a"], "b" => "c")
+        @test y === x
+
+        y = testf(replace!, Union{String, Missing}, x, ["a", "c", "", "a"], missing => "")
+        @test y === x
+    end
+
+    @testset "int to float" begin
+        x = categorical([0, 2, 0])
+
+        @testset "replace" begin
+            testf(replace, Float64, x, [0.5, 1.5, 0.5], 2 => 1.5, 0 => 0.5)
+            testf(replace, Float64, x, [0, 1, 0], 2 => 1.0)
+        end
+
+        @testset "replace!" begin
+            @test_throws InexactError replace!(x, 2 => 1.5)
+            if VERSION >= v"0.7.0-"
+                @test_throws InexactError replace!(Array(x), 2 => 1.5)
```

Comment:
This isn't the place to test how `replace!` works for `Array`. :-)

Reply:
Of course it isn't :)

my intent was to test that it behaves like replace! even when with invalid arguments. How would you test that? Or would you say that we don't need to test this and skip this test_throws?